ui/app/models/sync/destinations/azure-kv.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import SyncDestinationModel from '../destination';
+import { attr } from '@ember-data/model';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+const displayFields = ['name', 'keyVaultUri', 'tenantId', 'cloud', 'clientId', 'clientSecret'];
+const formFieldGroups = [
+  { default: ['name', 'keyVaultUri', 'tenantId', 'cloud', 'clientId'] },
+  { Credentials: ['clientSecret'] },
+];
+@withFormFields(displayFields, formFieldGroups)
+export default class SyncDestinationsAzureKeyVaultModel extends SyncDestinationModel {
+  @attr('string', {
+    label: 'Key Vault URI',
+    subText:
+      'URI of an existing Azure Key Vault instance. If empty, Vault will use the KEY_VAULT_URI environment variable if configured.',
+    editDisabled: true,
+  })
+  keyVaultUri;
+
+  @attr('string', {
+    label: 'Client ID',
+    subText:
+      'Client ID of an Azure app registration. If empty, Vault will use the AZURE_CLIENT_ID environment variable if configured.',
+  })
+  clientId;
+
+  @attr('string', {
+    subText:
+      'Client secret of an Azure app registration. If empty, Vault will use the AZURE_CLIENT_SECRET environment variable if configured.',
+  })
+  clientSecret; // obfuscated, never returned by API
```

Comment:
Same with this. Really any secrets should be masked, I would think.

Reply:
We don't mask them for OIDC config...but maybe that was built before we used that component